src/api/diskInstance.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import {join} from "path";
+import fs from "fs-extra";
+import {type PluginId} from "./pluginId";
+import {type Plugin} from "./plugin";
+import {rawParser} from "./instanceConfig";
+import {type TaskReporter, SilentTaskReporter} from "../util/taskReporter";
+
+export class DiskInstance {
+  +_instanceDirectory: string;
+  +_plugins: $ReadOnlyMap<PluginId, Plugin>;
+  +_taskReporter: TaskReporter;
+
+  constructor(
+    instanceDirectory: string,
+    plugins: $ReadOnlyMap<PluginId, Plugin>,
+    // The TaskReporter that will receive task timing info from Plugin methods.
+    // Defaults to the SilentTaskReporter.
+    taskReporter: ?TaskReporter
+  ) {
+    this._instanceDirectory = instanceDirectory;
+    this._plugins = plugins;
+    this._taskReporter = taskReporter || new SilentTaskReporter();
+  }
+
+  /**
+   * Gets the filesystem directory containing this Instance.
+   */
+  instanceDirectory(): string {
```

Comment:
For stuff like this I usually like to mark it as a [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) so its clear that you aren't performing any action:
```suggestion
  get instanceDirectory(): string {
```